Makefile

```diff
@@ -44,6 +44,8 @@ APPLICATION_DIR    := $(CURDIR)
 APPLICATION_NAME   := moltres
 BUILD_EXEC         := yes
 DEP_APPS           := $(shell $(FRAMEWORK_DIR)/scripts/find_dep_apps.py $(APPLICATION_NAME))
+ADDITIONAL_INCLUDES = -I/opt/moose/miniconda/include
+EXTERNAL_FLAGS = -I/opt/moose/miniconda/include -D_FORTIFY_SOURCE=2 -O2 -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I/opt/moose/miniconda/include   -L/opt/moose/miniconda/lib /opt/moose/miniconda/lib/libhdf5_hl_cpp.so /opt/moose/miniconda/lib/libhdf5_cpp.so /opt/moose/miniconda/lib/libhdf5_hl.so /opt/moose/miniconda/lib/libhdf5.so -L/opt/moose/miniconda/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,-rpath,/opt/moose/miniconda/lib -L/opt/moose/miniconda/lib -lrt -lpthread -lz -ldl -lm -Wl,-rpath -Wl,/opt/moose/miniconda/lib
```

Comment:
This will make it required for people to use conda with MOOSE. Not everyone will do this, so you should probably add some logic here on using -I/opt/moose/miniconda, i.e., only use it if it's there.

Reply:
Indeed. It's best to use an environment variable for this (like HDF5_ROOT and then use $(shell $(HDF5_ROOT)) everywhere.) 

Broader question @dogrady2 : Are you aiming for all moltres simulations to use this, or only some? These flags indicate that you expect all moltres simulations to use it. But, if that's the case, then all MoltresMaterials should be H5MoltresMaterials and you don't need two different classes. If both options are allowable operational modes, then the default should be the backwards compatible operational mode (no hdf5).

Reply:
I was aiming for all moltres simulations to use this. I felt that the MoltresMaterials source was difficult to follow and contained redundancies, for example each interpolation type had its own compute function defined. I can modify the H5MoltresMaterials such that it can support backward compatibility and read in the text format of the XS. I will look into the environment variable for the make file. I was having trouble initially with compiling, there is probably a better way to find the HDF5 library.

Reply:
Gotcha.

So, regarding finding hdf5...  my general tendency is to use something like CMake when creating a sophisticated build system -- it helps find packages on lots of different systems.  But -- I didn't write MOOSE, so they don't use CMake. The moose framework is a little home-baked, which is fine, but it means we need to follow their conventions and patterns... Generally, all the system specifics are handled with env vars and defaul values in the .mk files... https://github.com/idaholab/moose/blob/next/framework/build.mk

There are lots of ways to do this, but since Moltres wants to run on the big machines, making sure that we use the right hdf5 will be really important (particularly if it's reading/writing often... we may inadvertently slow moltres waaaaay down if this isn't done really carefully). Generally speaking, configuration files and environment variables seem to be the name of the game in the MOOSE universe, so let's start there.

Also, for this to pass the tests, the CI system is going to need hdf5.